secant/Features/Root/RootInitialization.swift

```diff
@@ -168,14 +183,40 @@ extension RootReducer {
                 return .none
 
             case .initialization(.nukeWallet):
-                walletStorage.nukeWallet()
-                do {
-                    try databaseFiles.nukeDbFilesFor(zcashSDKEnvironment.network)
-                } catch {
-                    // TODO: [#221] error we need to handle, issue #221 (https://github.com/zcash/secant-ios-wallet/issues/221)
+                guard sdkSynchronizer.isInitialized() else {
+                    // synchronizer is not available yet, start the app again and let checks to handle the rest
+                    state.isWipeRequested = true
+                    return EffectTask(value: .initialization(.checkWalletInitialization))
+                }
+                if !sdkSynchronizer.isSyncing() {
+                    // if not syncing, call the wipe on the synchronizer
+                    return EffectTask(value: .initialization(.nukeWalletAfterSynchronizerStops))
```

Comment:
you should stop the synchronizer anyway because it might start syncing because the timer fires between this and the next effect

Reply:
totally reworked the wipe(), no longer needed to deal with state of the syncrhonizer